### ISSUE TITLE
Select/Autocomplete features proposals list

### ISSUE THREAD

**Issue by 32penkin** [akveo/react-native-ui-kitten#801]
## 🚀 Feature Proposal
It would be great to add the following abilities:
_Props:_
- caption
- popover properties (at least placement)
- onBlur (needs discussion)
- onFocus (needs discussion)

_Methods:_
- focus
- blur

Note: some discussion related to implementation __focus__ and __blur__ methods is required.

## Motivation

This will help a lot in using these components in validation forms.

## Example

```tsx
<View>
        <Input
           label='Input'
           value={this.state.value1}
           onChangeText={this.setValue1}
           onSubmitEditing={this.onAutocompleteFocus}
        />
        <Autocomplete
           label='Autocomplete'
           ref={this.autocompleteRef}
           value={this.state.value2}
           placement='top'
           onChangeText={this.setValue2}
           onSelect={this.onSelectSomething1}
           onSubmitEditing={this.onSelectFocus}
        />
        <Select
           label='Select'
           ref={this.selectRef}
           data={this.selectData}
           caption={this.getCaption(errors, touched)}
           selectedOption={this.state.selectOption}
           onSelect={this.selectSomething2}
           onBlur={this.setSelectTouched}
        />
</View>
```

**Comment by gianpaj** [comment]
Hi, 

First of all, thanks for implementing #840 to programatically set the state of input components (`focus()`, `blur()`, `isFocused()`, `clear()`)

If I read the code correctly, it's not possible to bind a function on an `onBlur` or `onFocus` event on Select component since [TouchableOpacityProps](https://devdocs.io/react_native/touchablewithoutfeedback) doesn't have those, is that correct?

If so, could we keep this issue open? or, if preferred, open another one.

🙏

**Comment by artyorsh** [comment]
@gianpaj yes. In terms of `Select` focus and blur events mean exactly what the documentation says about it:
> Focuses Select and sets options list visible

Do we still missing some features here?

**Comment by gianpaj** [comment]
@artyorsh cheers, but I mean to be able to **listen** on blur events. i.e. I have the Select "dropdown menu" opened and I want to run a function when somebody clicks outside the Select and closes it.

![Jan-28-2020 13-20-26](https://user-images.githubusercontent.com/899175/73267640-5cda4480-41d1-11ea-830f-e1fa223f3f18.gif)


I'm trying to check the number of selected options as soon as the user clicks outside as I would like to limit for example to 2

**Comment by artyorsh** [comment]
Got your idea, can you please move your issue for this? I guess it could be available in 4.4. Thanks👍

**Comment by junniepat** [comment]
How can one get the value of the selected option not row id. I need this.
